x-pack/elastic-agent/pkg/agent/cmd/container.go

```diff
@@ -207,18 +209,24 @@ func containerCloudCmd(streams *cli.IOStreams, cmd *cobra.Command, flags *global
 	return nil
 }
 
-func containerCmd(streams *cli.IOStreams, cmd *cobra.Command, flags *globalFlags, cfg setupConfig) error {
+func containerCmd(streams *cli.IOStreams, cmd *cobra.Command, cfg setupConfig) error {
 	var err error
 	var client *kibana.Client
 	executable, err := os.Executable()
 	if err != nil {
 		return err
 	}
 
+	// set paths early so all action below use the defined paths
+	err = handlePaths(cfg.Agent.Paths)
```

Comment:
The APM Server is extracted from the Elastic Agent's download folder. At the moment this is done before the Elastic Agent gets started (https://github.com/elastic/beats/pull/24817/files#diff-1f8e1e2079524931fa8f3cf121667a25baa2b998557136500d5e7596b46e9187R159), so also before you are setting the path variables here. The path variables need to be set as the very first action.

If you prefer I can change the order to make this work with the cloud mode after your PR gets merged.

Reply:
I think because of me pushing an update I broke the link your provided. Can you provide it again so I can see what you are talking about. Would be happy to fix it in this branch.

Reply:
certainly; https://github.com/elastic/beats/blob/master/x-pack/elastic-agent/pkg/agent/cmd/container.go#L157